### PR TITLE
DOC: add two new examples to the transform docs

### PR DESCRIPTION
**Pull request by marktgraham**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This PR adds two examples to the transform user guide. The examples apply the transform method to a groupby object. This is from the PyData Global sprint, from [this issue](https://github.com/MarcoGorelli/PyDataGlobal2020-sprint/issues/5). The examples are taken from [here](https://stackoverflow.com/a/37189976/14583166) and [here](https://stackoverflow.com/a/54417351/14583166).

### CODE REVIEW

**Comment by marktgraham** [comment]
Here is a preview of the new examples:
Screen Shot 2020-11-17 at 17 20 15
Screen Shot 2020-11-17 at 17 20 24

File: pandas/core/shared_docs.py [hunk 1]
```diff
@@ -324,4 +324,57 @@
 0  0.000000   1.000000
 1  1.000000   2.718282
 2  1.414214   7.389056
+
+It is possible to call transform on a GroupBy object:
```
Comment (jreback) [inline pandas/core/shared_docs.py:328]: You can call transform on a GroupBy object.

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, there are some pre commit errors, can you merge master and ping on green

File: pandas/core/shared_docs.py (hunk 1 above)
Comment (marktgraham) [reply pandas/core/shared_docs.py]: I have made the change.

**Comment by marktgraham** [comment]
@jreback I have merged with master and now all tests pass.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @marktgraham

**Pull request merged**